cinnamonvantage@garlayntoji/files/cinnamonvantage@garlayntoji/applet.js

```diff
@@ -0,0 +1,249 @@
+                this.winkeySwitch.setToggleState(state);
+                break;
+            case TOUCHPAD:
+                this.touchpadSwitch.setToggleState(state);
+                break;
+            case CONSERVATION_MODE:
+                this.conservationModeSwitch.setToggleState(state);
+                break;
+            case RAPID_CHARGE:
+                this.rapidChargeSwitch.setToggleState(state);
+                break;
+            case USB_CHARGING:
+                this.usbChargingSwitch.setToggleState(state);
+                break;
+            case OVERDRIVE:
+                this.overdriveSwitch.setToggleState(state);
+                break;
+            case GSYNC:
+                this.gsyncSwitch.setToggleState(state);
+                break;
+        }
+    },
+
+    // Changing driver file's content upon switch toggle event from dropdown menu
+    _onSwitchToggled: function(item, state, key) {
+        let previousState = this.settings.getValue(key);
+        let filePath = FILE_PATHS[key];
+        let value = state ? "1" : "0";
+        let command = `pkexec sh -c "echo '${value}' > ${filePath}"`;
+
+        Util.spawnCommandLineAsyncIO(command, (stdout, stderr, exitCode) => {
+            if (exitCode === 0) {
+                // Command succeeded, update the state in settings
+                this.settings.setValue(key, state);
+                if (key === GSYNC) {
+                    this._showRestartNotification();
+                }
+            } else {
+                // Command failed, revert the switch state
+                this._updateSwitchState(key, previousState);
+                globalThis.logError(`Failed to write to file ${filePath}: ${stderr}`);
+            }
+        });
+    },
+
+    // Changing driver file's content upon switch toggle event from settings page
+    _onSettingsChanged: function(settings, key) {
+        let state = this.settings.getValue(key);
+        let filePath = FILE_PATHS[key];
+        let value = state ? "1" : "0";
+        let command = `pkexec sh -c "echo '${value}' > ${filePath}"`;
+
+        Util.spawnCommandLineAsyncIO(command, (stdout, stderr, exitCode) => {
+            if (exitCode === 0) {
+                // Command succeeded, update the state in the applet
+                this._updateSwitchState(key, state);
+                if (key === GSYNC) {
+                    this._showRestartNotification();
+                }
+            } else {
+                // Command failed, revert the switch state
+                let previousState = !state;
+                this.settings.setValue(key, previousState);
+                globalThis.logError(`Failed to write to file ${filePath}: ${stderr}`);
+            }
+        });
+    },
+
+    _showRestartNotification: function() {
+        let command = `notify-send "CinnamonVantage: system reboot required" "A system reboot is required for this change to take effect."`
+        Util.spawnCommandLineAsync(command)
```

Comment:
Again, this works but you can use the MessageTray module rather than having to exec a dependency and spawn a shell.